### DISCUSSION TITLE
Splunk HEC Sink support for channel parameter

### DISCUSSION THREAD

**Discussion by mans2singh** [vectordotdev/vector#14773] [Q&A] [answered]
Hi:

[Splunk HEC collector](https://docs.splunk.com/Documentation/Splunk/8.2.6/RESTREF/RESTinput#services.2Fcollector) supports a `channel` parameter.  In Vector's [Splunk Sink](https://master.vector.dev/docs/reference/configuration/sinks/splunk_hec_logs/) docs, I could not find this parameter.  

Please let me know if this will be supported in the future and whether you have any recommendation on how to pass it with the current implementation.

Thanks

**Comment by spencergilbert** [comment, marked as answer]
👋 

Hi @mans2singh,

Are you referring to the "channel" used for: https://docs.splunk.com/Documentation/Splunk/8.2.3/Data/AboutHECIDXAck

If so, that's configurable via: https://master.vector.dev/docs/reference/configuration/sinks/splunk_hec_logs/#acknowledgements and documented further at https://master.vector.dev/docs/reference/configuration/sinks/splunk_hec_logs/#indexer-acknowledgements

**Reply by mans2singh** [reply]
Yes @spencergilbert that's the Splunk HEC channel attribute.  

From your reference it looks like the sink will generate a random channel GUID and use it internally to query for acknowledgements.
So, we don't need set up the GUID if we set [acknowledgements.enabled](https://master.vector.dev/docs/reference/configuration/sinks/splunk_hec_logs/#acknowledgements.enabled) to true.

Thanks.
